drivers/timer/riscv_machine_timer.c

```diff
@@ -96,6 +104,15 @@ static void set_mtimecmp(uint64_t time)
 #endif
 }
 
+static void set_divider(uint32_t divider)
+{
+#ifdef MTIMEDIV_REG
```

Comment:
This seems clumsy.  Would be better to predicate on the same DTS node you use above, or use the node to set a "HAVE_DIVIDER" symbol or something to use.

Reply:
Hi @andyross , could you please reply?

Reply:
Hi @andyross, kind reminder (please see above)

Reply:
Sorry, coming back.  This was just a code style thing.  Using "is this low level macro that happens to store the address of a MMIO register defined?" as a proxy for "should I attempt MMIO access?" is sort of clumsy.  In general questions of the form "does the hardware have this feature?" should be answered by devicetree.

So yes, I'd view either of your two choices as preferable.  But my +1 on this PR stands regardless.

Reply:
Thanks! Done